tools/zypp-CheckAccessDeleted.cc

```diff
@@ -94,33 +94,59 @@ struct ProcInfoTable
   TableCol files;
 };
 
+void usage ( const std::string &appname )
+{
+  std::cout << "Usage: " << appname << " [--help]" << std::endl;
+  std::cout << "List information about all running processe" << std::endl;
+  std::cout << "which access deleted executables or libraries." << std::endl;
+  std::cout << "  PID     " << "process ID" << std::endl;
+  std::cout << "  PPID    " << "parent process ID" << std::endl;
+  std::cout << "  UID     " << "process user ID" << std::endl;
+  std::cout << "  LOGIN   " << "process login name" << std::endl;
+  std::cout << "  COMMAND " << "process command name" << std::endl;
+  std::cout << "  SERVICE " << "/etc/init.d/ script that might be used to restart the command (guessed)" << std::endl;
+  std::cout << "  FILES   " << "list of deleted executables or libraries accessed" << std::endl;
+}
+
 int main( int argc, char * argv[] )
 {
-  if ( argc >= 2 )
-  {
-    std::string progname( zypp::Pathname::basename( argv[0] ) );
-    if ( strcmp( argv[1], "--help" ) == 0 )
+
+  std::string progname( zypp::Pathname::basename( argv[0] ) );
+  std::string debugInputFile;
+  argv++;
+  argc--;
+
+  while( argc > 0 ) {
+    if ( strcmp( argv[0], "--help" ) == 0 )
     {
-      std::cout << "Usage: " << progname << " [--help]" << std::endl;
-      std::cout << "List information about all running processe" << std::endl;
-      std::cout << "which access deleted executables or libraries." << std::endl;
-      std::cout << "  PID     " << "process ID" << std::endl;
-      std::cout << "  PPID    " << "parent process ID" << std::endl;
-      std::cout << "  UID     " << "process user ID" << std::endl;
-      std::cout << "  LOGIN   " << "process login name" << std::endl;
-      std::cout << "  COMMAND " << "process command name" << std::endl;
-      std::cout << "  SERVICE " << "/etc/init.d/ script that might be used to restart the command (guessed)" << std::endl;
-      std::cout << "  FILES   " << "list of deleted executables or libraries accessed" << std::endl;
+      usage( progname );
       return 0;
+    } else if ( strcmp( argv[0], "--fromDebugFile" ) ) {
```

Comment:
Option should be mentioned in `usage()`

    ma@fibonacci:~ (0) $ zypp-CheckAccessDeleted --fromDebugFile ~/bsc1079991.lsof
    zypp-CheckAccessDeleted: unexpected argument '--fromDebugFile'
    Try `zypp-CheckAccessDeleted --help' for more information.

    ma@fibonacci:~ (1) $ zypp-CheckAccessDeleted --fromDebugFile
    zypp-CheckAccessDeleted: unexpected argument '--fromDebugFile'
    Try `zypp-CheckAccessDeleted --help' for more information.

    ma@fibonacci:~ (1) $ zypp-CheckAccessDeleted --fromDebugFil
    zypp-CheckAccessDeleted: fromDebugFile requires a argument

Untested? (strcmp() == 0)

Reply:
Using this together with zypper, I'd almost vote for using the same option name `--debugFile` in both cases.